src/controllers/auth.js

```diff
@@ -1,2 +1,62 @@
 const bcrypt = require('bcrypt');
 const UserModel = require('../models/user');
+const { generateUniqeUsername } = require('../services/utils');
+
+// eslint-disable-next-line consistent-return
+const register = async (req, res) => {
+  const errorsArray = [];
+  const {
+    firstName,
+    lastName,
+    email,
+    phoneNumber,
+    age,
+    gender,
+    nationality,
+    refugee,
+    password0,
+  } = req.body;
+
+  try {
+    const usedEmail = await UserModel.findOne({ email });
+    if (usedEmail) {
+      errorsArray.push('Email is already taken');
+    }
+
+    if (errorsArray.length > 0) {
```

Comment:
I dont see the usage of this, like before it there's only 1 error that u are checking with and its basicaly returning it as json, so I think this arr can be removed and u can have smth like this
```
if (usedEmail) {
  return res.status(400).json({error:"Email is already taken"})
}
```

Reply:
I totally forgot about that array. Now fixed.